charts/kyverno/templates/configmap.yaml

```diff
@@ -9,16 +9,32 @@ metadata:
 data:
   # resource types to be skipped by kyverno policy engine
   {{- if .Values.config.resourceFilters }}
-  resourceFilters: {{ tpl (join "" .Values.config.resourceFilters) . | quote }}
+    {{- $resourceFilters := .Values.config.resourceFilters }}
+    {{- if .Values.excludeKyvernoNamespace }}
+      {{- $resourceFilters = prepend .Values.config.resourceFilters (printf "[*,%s,*]" (include "kyverno.namespace" .)) }}
+    {{- end }}
+  resourceFilters: {{ tpl (join "" $resourceFilters) . | quote }}
   {{- end -}}
   {{- if .Values.config.excludeGroupRole }}
   excludeGroupRole: {{ join "," .Values.config.excludeGroupRole | quote }}
   {{- end -}}
   {{- if .Values.config.excludeUsername }}
   excludeUsername: {{ join "," .Values.config.excludeUsername | quote }}
   {{- end -}}
-  {{- if .Values.config.webhooks }}
-  webhooks: {{ .Values.config.webhooks  | toJson | quote }}
+  {{- if or .Values.config.webhooks }}
+    {{- $excludeDefault := dict "key" "kubernetes.io/metadata.name" "operator" "NotIn" "values" (list (include "kyverno.namespace" .)) }}
+    {{- $newWebhook := list }}
+    {{- range $webhook := .Values.config.webhooks }}
```

Comment:
won't work if the initial list is empty

Reply:
How so?  There is the `else` block that defines the static exclusion if webhooks are not defined and `excludeKyvernoNamespace` is true.

Reply:
Yeah, didn't see it at first sight.
Wouldn't it be simpler to add the logic at the top of the file feeding a `$webhooks` variable, and use that in the CM template ?
I find it harder to read when logic and templating are interleaved.

Reply:
My whole template:
```
{{- if (not .Values.config.existingConfig) }}
{{- $webhooks := .Values.config.webhooks }}
{{- if .Values.excludeKyvernoNamespace }}
{{- $webhooks = default (list dict) $webhooks }}
{{- $excludeDefault := dict "key" "kubernetes.io/metadata.name" "operator" "NotIn" "values" (list (include "kyverno.namespace" .)) }}
{{- range $_, $webhook := $webhooks }}
{{- $namespaceSelector := default dict $webhook.namespaceSelector }}
{{- $matchExpressions := default list $namespaceSelector.matchExpressions }}
{{- $newNamespaceSelector := dict "matchLabels" $namespaceSelector.matchLabels "matchExpressions" (append $matchExpressions $excludeDefault) }}
{{- $newWebhook := merge (omit $webhook "namespaceSelector") (dict "namespaceSelector" $newNamespaceSelector) }}
{{- $webhooks = append $webhooks $newWebhook }}
{{- end }}
{{- end }}
apiVersion: v1
kind: ConfigMap
metadata:
  labels: {{ include "kyverno.labels" . | nindent 4 }}
    app: kyverno
  name: {{ template "kyverno.configMapName" . }}
  namespace: {{ template "kyverno.namespace" . }}
data:
  # resource types to be skipped by kyverno policy engine
  {{- if .Values.config.resourceFilters }}
  resourceFilters: {{ tpl (join "" .Values.config.resourceFilters) . | quote }}
  {{- end -}}
  {{- if .Values.config.excludeGroupRole }}
  excludeGroupRole: {{ join "," .Values.config.excludeGroupRole | quote }}
  {{- end -}}
  {{- if .Values.config.excludeUsername }}
  excludeUsername: {{ join "," .Values.config.excludeUsername | quote }}
  {{- end -}}
  {{- if $webhooks }}
  webhooks: {{ $webhooks | toJson | quote }}
  {{- end -}}
  {{- if .Values.config.generateSuccessEvents }}
  generateSuccessEvents: {{ .Values.config.generateSuccessEvents | quote }}
  {{- end -}}
{{- end -}}
```